crates/cranelift/src/debug/transform/unit.rs

```diff
@@ -367,6 +389,11 @@ where
             continue;
         }
 
+        if is_dead_code(entry) {
+            skip_children = true;
```

Comment:
I'm not personally really all that familiar with this code, but it seems to me like `skip_children` should be reset back to `false` at some point after all children have been skipped?

Reply:
Ugh, of course. I'll fix that and update the test to make sure it's done properly.